tests/Api/ProductMediaFile/ListProductMediaFileApiIntegration.php

```diff
@@ -0,0 +1,214 @@
+
+        $lastPage = $secondPage->getNextPage();
+
+        $this->assertInstanceOf(PageInterface::class, $lastPage);
+        $this->assertTrue($lastPage->hasPreviousPage());
+        $this->assertFalse($lastPage->hasNextPage());
+        $this->assertNull($lastPage->getNextLink());
+        $this->assertNull($lastPage->getNextPage());
+        $this->assertCount(0, $lastPage->getItems());
+
+        $previousPage = $lastPage->getPreviousPage();
+        $this->assertInstanceOf(PageInterface::class, $secondPage);
+
+        $mediaFiles = $previousPage->getItems();
+
+        $this->assertCount(2 ,$mediaFiles);
+        $this->assertSameContent($expectedMediaFiles[2], $this->sanitizeMediaFile($mediaFiles[0]));
+        $this->assertSameContent($expectedMediaFiles[3], $this->sanitizeMediaFile($mediaFiles[1]));
+    }
+
+    public function testListPerPageWithCount()
+    {
+        $api = $this->createClient()->getProductMediaFileApi();
+        $baseUri = $this->getConfiguration()['api']['baseUri'];
+
+        $firstPage = $api->listPerPage(2, true);
+        $this->assertInstanceOf(PageInterface::class, $firstPage);
+        $this->assertSame(4, $firstPage->getCount());
+        $this->assertSame($baseUri . '/api/rest/v1/media-files?page=2&limit=2&with_count=true', $firstPage->getNextLink());
+    }
+
+    public function testListPerPageWithSpecificQueryParameter()
+    {
+        $api = $this->createClient()->getProductMediaFileApi();
+        $baseUri = $this->getConfiguration()['api']['baseUri'];
+        $expectedMediaFiles = $this->getExpectedMediaFiles();
+
+        $firstPage = $api->listPerPage(2, false, ['foo' => 'bar']);
+        $this->assertInstanceOf(PageInterface::class, $firstPage);
+        $this->assertSame($baseUri . '/api/rest/v1/media-files?page=2&limit=2&with_count=false&foo=bar', $firstPage->getNextLink());
+
+        $mediaFiles = $firstPage->getItems();
+
+        $this->assertCount(2 ,$mediaFiles);
+        $this->assertSameContent($expectedMediaFiles[0], $this->sanitizeMediaFile($mediaFiles[0]));
+        $this->assertSameContent($expectedMediaFiles[1], $this->sanitizeMediaFile($mediaFiles[1]));
+    }
+
+    public function testAll()
+    {
+        $api = $this->createClient()->getProductMediaFileApi();
+        $mediaFiles = $api->all();
+        $expectedMediaFiles = $this->getExpectedMediaFiles();
+
+        $this->assertInstanceOf(ResourceCursorInterface::class, $mediaFiles);
+
+        $mediaFilesCount = 0;
+        foreach ($mediaFiles as $key => $mediaFile) {
+            $this->assertSameContent($expectedMediaFiles[$key], $this->sanitizeMediaFile($mediaFile));
+            $mediaFilesCount++;
+        }
+
+        $this->assertSame(4, $mediaFilesCount);
+    }
+
+    public function testAllWithUselessQueryParameter()
+    {
+        $api = $this->createClient()->getProductMediaFileApi();
+        $mediaFiles = $api->all(10, ['foo' => 'bar']);
```

Comment:
Nope. The useless parameter appear only in pages URI and they are hidden in `ResourceCursor` object